### PR TITLE
chore: use tracing instead of log

### LINKED ISSUE #3095: Remove the `log` crate
**Issue summary**

Just use `tracing`

**Other information and links**

### PR DESCRIPTION
**Pull request by creativcoder**
## Summary of changes

Changes introduced in this pull request:

- Removes the log crate and replaces with tracing which is already used.

## Reference issue to close (if applicable)

Closes #3095 

## Change checklist

- [ ] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation,
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by aatifsyed** [comment]
Love this 😍

File: src/cli_shared/cli/config.rs [hunk 1]
```diff
@@ -1,18 +1,125 @@
...
-use serde::{Deserialize, Serialize};
+use core::time::Duration;
+use serde::{
+    de::{DeserializeSeed, EnumAccess, Error, Unexpected, VariantAccess, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::{fmt, path::PathBuf, str::FromStr, sync::Arc};
+use tracing::log::LevelFilter;
 
 use super::client::Client;
```
Comment (aatifsyed) [inline src/cli_shared/cli/config.rs:17]: This doesn't look right. Why do we need to serialize log levels? Maybe removing that should be a separate PR.

For now, let's just
```
#[derive(serde::Serialize, serde::Deserialize)]
#[serde(rename = ...)]
enum LogLevel {
    Off,
    ...
}
```

**Review by aatifsyed — APPROVED** [review]

**Review by lemmih — APPROVED** [review]
